drivers/pwm/pwm_npcx.c

```diff
@@ -212,21 +212,26 @@ static int pwm_npcx_init(const struct device *dev)
 	pwm_npcx_configure(dev, config->clk_cfg.bus);
 
 	/* Configure pin-mux for PWM device */
-	npcx_pinctrl_mux_configure(config->alts_list, config->alts_size, 1);
+	ret = pinctrl_apply_state(config->pcfg, PINCTRL_STATE_DEFAULT);
+	if (ret < 0) {
+		LOG_ERR("PWM pinctrl setup failed (%d)", ret);
+		return ret;
+	}
 
 	return 0;
 }
 
+#define NPCX_PWM_OPEN_DRAIN(inst) \
+	DT_PROP(DT_INST_PHANDLE(inst, pinctrl_0), drive_open_drain)
```

Comment:
this needs to be handled by a pin control driver, not like this. If some pin properties are handled directly by certain IP blocks (like e.g. Nordic), you can either encode address/bit in the state node or make use of the `PINCTRL_STORE_REG`, so that you also get the peripheral address when being called.

Reply:
Got it. Will modify it for code review later.

Reply:
Done. The latest PR has moved sources that configure ```drive-open-drain``` from pwm_npcx.c to pinctrl_npcx.c. In order to make sure static casting of ```reg``` is reliable, a new property, ```drive-supported```, is introduced in pwm pinctrl rebuild nodes.